src/FFmpegWriter.cpp

```diff
@@ -1283,13 +1296,6 @@ AVStream *FFmpegWriter::add_video_stream() {
 	st->avg_frame_rate = av_inv_q(c->time_base);
 	st->time_base.num = info.video_timebase.num;
 	st->time_base.den = info.video_timebase.den;
-#if (LIBAVFORMAT_VERSION_MAJOR >= 58)
-	#pragma GCC diagnostic push
-	#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-	st->codec->time_base.num = info.video_timebase.num;
-	st->codec->time_base.den = info.video_timebase.den;
-	#pragma GCC diagnostic pop
-#endif
```

Comment:
This is removed because you can't assign to values in the _codec_ — it's now `const`ed, so even the deprecation-override pragma wouldn't cut it anymore. This was never necessary, whatever it was trying to achieve can't be done the way it was trying to do it. Modifying the `AVFormat`, the `AVCodec`, or any other ffmpeg structure that's not one of the corresponding **contexts** (`AVFormatContext`, `AVCodecContext`, etc.) has always been unsupported, and with the expanded `const`-correctness of the latest library versions, is no longer possible.